plugins/of-the-day/manager.py

```
@@ -389,6 +393,31 @@ def _wrap_text(self, text: str, max_width: int, font, max_lines: int = 10) -> Li
     
+    def _fit_title(self, title: str, font) -> str:
+        """Ellipsize the title to the panel width.
+
+        Titles that already fit are returned unchanged, so normal-width panels
+        render exactly as before; on a narrow panel (64px) a long word is
+        truncated with '...' instead of being drawn past the panel edge.
+        """
+        def _w(text: str) -> int:
+            try:
+                return self.display_manager.get_text_width(text, font)
+            except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)